File: 414-third-maximum-number/third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
        TreeSet<Integer> uniqueNums = new TreeSet<>(Collections.reverseOrder());
        
       
        for (int num : nums) {
            uniqueNums.add(num);
        }
        
        List<Integer> sortedList = new ArrayList<>(uniqueNums);
        
        
        if (sortedList.size() < 3) {
            return sortedList.get(0);
        } else {
            return sortedList.get(2); 
        }
    }
}

File: 414-third-maximum-number/third-maximum-number.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
         p=sorted(nums)
         a=[]
         for i in p:
            if i not in a:
                 a.append(i)
       
         if len(a)== 2:
             return a[-1]
         elif len(a)==1:
             return a[0]
         else:
             return a[-3]